### PR TITLE
14.0 mig product configurator

### PR DESCRIPTION
**Pull request by dreispt**
Migrate to 14.0 base `product_configurator` module, with history.
Precommit applied, two issues need fixing (need help):

- product_configurator/models/product_config.py:1254:5: C901 'ProductConfigSession.validate_configuration' is too complex (18)
- product_configurator/wizard/product_configurator.py:584:5: C901 'ProductConfigurator.add_dynamic_fields' is too complex (23)

### CODE REVIEW

**Comment by Chandresh-OSI** [comment]
@dreispt Ready for the review.
CC @bizzappdev @PCatinean

**Comment by dreispt** [comment]
@Chandresh-OSI Doing sanity checks, I get an error when opening the Product Configurator with the demo data installed: 
image

**Comment by dreispt** [comment]
I also see a few warnings on the Log file. Can these be fixed please?
```
2021-04-20 17:24:40,776 16186 WARNING 14-boti-0419 odoo.fields: Field product.attribute.custom_type: unknown parameter 'size', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
2021-04-20 17:24:40,816 16186 WARNING 14-boti-0419 odoo.fields: Field product.config.line.value_ids: unknown parameter 'id1', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
2021-04-20 17:24:40,816 16186 WARNING 14-boti-0419 odoo.fields: Field product.config.line.value_ids: unknown parameter 'id2', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
2021-04-20 17:24:40,887 16186 INFO 14-boti-0419 odoo.modules.registry: module product_configurator: creating or updating database tables
2021-04-20 17:24:41,001 16186 WARNING 14-boti-0419 odoo.addons.base.models.ir_model: Two fields (name, config_name) of product.product() have the same label: Name.
2021-04-20 17:24:41,187 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/security/configurator_security.xml
2021-04-20 17:24:41,822 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/views/res_config_settings_view.xml
2021-04-20 17:24:42,068 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/data/menu_configurable_product.xml
2021-04-20 17:24:42,173 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/data/product_attribute.xml
2021-04-20 17:24:42,186 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/data/ir_sequence_data.xml
2021-04-20 17:24:42,193 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/data/ir_config_parameter_data.xml
2021-04-20 17:24:42,199 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/security/ir.model.access.csv
2021-04-20 17:24:42,557 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/views/assets.xml
2021-04-20 17:24:42,680 16186 INFO 14-boti-0419 odoo.modules.loading: loading product_configurator/views/product_view.xml
2021-04-20 17:24:43,055 16186 WARNING 14-boti-0419 odoo.models: @onchange('standard_price', 'activity_cost_ids', 'activity_cost_ids.standard_price') parameters must be field names
```

File: product_configurator/models/product_config.py [hunk 1]
```diff
@@ -207,8 +207,8 @@ def _compute_template_attribute_ids(self):
     )
     value_ids = fields.Many2many(
         comodel_name="product.attribute.value",
-        id1="cfg_line_id",
-        id2="attr_val_id",
+        column1="cfg_line_id",
+        column2="attr_val_id",
```
Comment (dreispt) [inline product_configurator/models/product_config.py:211]: @Chandresh-OSI Mind that this fix might break your dev/test database. To recover I had to uninstall and the reinstall the module.

**Comment by dreispt** [comment]
@Chandresh-OSI I fixed the issues I found

File: product_configurator/models/product_config.py (hunk 1 above)
Comment (Chandresh-OSI) [reply product_configurator/models/product_config.py:211]: Noted.

**Comment by dreispt** [comment]
@Chandresh-OSI I don't think the wizard should look like this.
I guess the "Attributes" field should not be visible, and the selection option should depend on the step we are at.
Can you check?

image

**Comment by Chandresh-OSI** [comment]
@dreispt wizard looks like very similar to 13.0. I will check once again although.

**Comment by dreispt** [comment]
Yes, "Attributes" is there in v13, but after that the Fuel/Engine/... fields are not in the initial page,

**Pull request merged**